store/postgres/tests/relational_bytes.rs

```diff
@@ -253,24 +275,29 @@ fn find_many() {
         insert_thing(conn, layout, ID, NAME);
         insert_thing(conn, layout, ID2, NAME2);
 
-        let mut id_map: BTreeMap<&EntityType, Vec<&str>> = BTreeMap::default();
-        id_map.insert(&*THING, vec![ID, ID2, "badd"]);
+        let mut id_map = BTreeMap::default();
+        id_map.insert(
+            (THING.clone(), CausalityRegion::ONCHAIN),
+            vec![ID.to_string(), ID2.to_string(), "badd".to_string()],
+        );
 
         let entities = layout
             .find_many(conn, &id_map, BLOCK_NUMBER_MAX)
             .expect("Failed to read many things");
-        assert_eq!(1, entities.len());
-
-        let ids = entities
-            .get(&*THING)
-            .expect("We got some things")
-            .iter()
-            .map(|thing| thing.id().unwrap())
-            .collect::<Vec<_>>();
-
-        assert_eq!(2, ids.len());
-        assert!(ids.contains(&ID.to_owned()), "Missing ID");
-        assert!(ids.contains(&ID2.to_owned()), "Missing ID2");
+        assert_eq!(2, entities.len());
+
+        let id_key = EntityKey {
+            entity_id: ID.into(),
+            entity_type: THING.clone(),
+            causality_region: CausalityRegion::ONCHAIN,
+        };
+        let id2_key = EntityKey {
+            entity_id: ID2.into(),
+            entity_type: THING.clone(),
+            causality_region: CausalityRegion::ONCHAIN,
```

Comment:
I guess these are the ones I missed earlier. maybe add one that's not ONCHAIN?

Reply:
It would be a bit annoying to test this here since it would be necessary to add to the test layout a new entity type that somehow has a causality region column. I added coverage for looking up offchain entities to the integration tests, and added a `debug_assert!` to the entity cache that ensures cache consistency (so the test is not looking up only on the cache).

Reply:
I think this is the price we have to pay for the design decision and current test strategy. IMO it's worth adding this coverage even if it takes longer to write. I would, however, add it in a separate PR.